src/Data/List/Lazy.purs

```diff
@@ -162,11 +162,28 @@ infix 8 range as ..
 
 -- | Create a list containing a range of integers, including both endpoints.
 range :: Int -> Int -> List Int
-range start end | start == end = singleton start
-                | otherwise = go end start (if start > end then 1 else -1) nil
+range start end
+    | start > end =
+        let g x | x >= end  = Just (Tuple x (x - 1))
```

Comment:
this weird little syntax wart was caused by a strange parse error -- will file an issue on the compiler (or maybe where doesn't work like I expected)